packages/cli/src/server/web/server.rs

```diff
@@ -81,9 +81,16 @@ pub async fn setup_router(
     ));
 
     router = if let Some(base_path) = config.dioxus_config.web.app.base_path.clone() {
-        let base_path = format!("/{}", base_path.trim_matches('/'));
+        // NB. The trailing `/` is required for Dioxus Routers to match on root paths. Without it,
+        // navigating to e.g. `/base/path/` results in the below fallback being hit, and users
+        // receiving a message that `/base/path/` is "Outside of the base path: /base/path".
+        // Navigating to e.g. `/base/path` results in the below fallback  _not_ being hit, but a
+        // panic getting thrown from `WebHistory::route_from_location`. Interestingly, the
+        // `RouteParseError` passed into the panic reports that attempts were made to match on no
+        // routes at all.
+        let base_path = format!("/{}/", base_path.trim_matches('/'));
```

Comment:
After looking at the router-macro source a bit more, I realize this is (probably?) intentional behavior. A [check](https://github.com/ealmloff/dioxus/blob/58304b7a69c123f842e05ab5814ef89a5d469619/packages/router-macro/src/lib.rs#L491) introduced in #1272 explicitly rejects all routes without a leading slash.

I'm not sure if there should be a special case in the generated `from_str` method for the `s.trim().len() == 0` case, or if I'm misunderstanding something about canonical URI construction?

Reply:
If I remember correctly, the check was introduced so that the route `href=bla` is not parsed as `href=/bla` in a link because browsers do not interpret it that way. Parsing an empty route we get from the browser as the base route seems unproblematic